sdk/test/Services/RDS/UnitTests/Custom/RDSAuthTokenGeneratorTest.cs

```diff
@@ -41,26 +45,34 @@ public class RDSAuthTokenGeneratorTest
         private static readonly AWSCredentials BasicCredentials = new BasicAWSCredentials(AccessKey, SecretKey);
         private static readonly AWSCredentials SessionCredentials = new SessionAWSCredentials(AccessKey, SecretKey, SessionToken);
 
-        private List<FallbackCredentialsFactory.CredentialsGenerator> originalFallbackList;
+        private Dictionary<Type, IIdentityResolver> originalIdentityResolvers;
 
         [TestInitialize]
         public void Initialize()
         {
-            originalFallbackList = FallbackCredentialsFactory.CredentialsGenerators;
+            AWSConfigs.AWSRegion = AWSRegion.SystemName;
+
+            FieldInfo field = typeof(DefaultIdentityResolverConfiguration).GetField
+                ("identityResolvers", BindingFlags.Static | BindingFlags.NonPublic);
```

Comment:
Adding `Reset` method will not allow us to change the resolvers, the only way to do that is to make them public similar to `FallbackCredentialsFactory.CredentialsGenerators`, this made sense for `FallbackCredentialsFactory` since we use it directly in the serviceClients and there is no way of overriding it, however we do allow users to override the identity resolvers and I don't think we should change make a field public just for the tests.